domains/client/domain-operator/src/tests.rs

```diff
@@ -1789,6 +1789,190 @@ async fn test_bundle_equivocation_fraud_proof() {
+        Alice,
+        BasePath::new(directory.path().join("alice")),
+    )
+    .build_evm_node(Role::Authority, GENESIS_DOMAIN_ID, &mut ferdie)
+    .await;
+    produce_blocks!(ferdie, alice, 1).await.unwrap();
+
+    // produce a transfer from alice to bob in one bundle, which should succeed
+    let pre_alice_free_balance = alice.free_balance(Alice.to_account_id());
+    let alice_account_nonce = alice.account_nonce();
+
+    let tx = alice.construct_extrinsic(
+        alice_account_nonce,
+        pallet_balances::Call::transfer_allow_death {
+            dest: Bob.to_account_id(),
+            // we send half of alice balance to bob
+            value: pre_alice_free_balance / 2,
+        },
+    );
+    alice
+        .send_extrinsic(tx)
+        .await
+        .expect("Failed to send extrinsic");
+
+    // Produce a bundle and submit to the tx pool of the consensus node
+    let (_slot, bundle) = ferdie.produce_slot_and_wait_for_bundle_submission().await;
+    assert!(bundle.is_some());
+    let bundle = bundle.unwrap();
+    assert_eq!(bundle.extrinsics.len(), 1);
+
+    // we produce another bundle with similar transaction
+    // second one will fail at pre dispatch since we use same nonce
+    // we change the value to something else so that ext is different
+    // send the tip so that previous ext is replaced with current in Domain tx pool
+    let tip = 1000000;
+    let tx = alice.construct_extrinsic_with_tip(
+        // use the same nonce
+        alice_account_nonce,
+        tip,
+        pallet_balances::Call::transfer_allow_death {
+            dest: Bob.to_account_id(),
+            value: pre_alice_free_balance / 3,
+        },
+    );
+    alice
+        .send_extrinsic(tx)
+        .await
+        .expect("Failed to send extrinsic");
+
+    // Produce a bundle and submit to the tx pool of the consensus node
+    let (slot, bundle) = ferdie.produce_slot_and_wait_for_bundle_submission().await;
+    assert!(bundle.is_some());
+    let bundle = bundle.unwrap();
+    assert_eq!(bundle.extrinsics.len(), 1);
+
+    // produce block and import domain block
+    produce_block_with!(ferdie.produce_block_with_slot(slot), alice)
+        .await
+        .unwrap();
+
+    // domain block body should have 3 extrinsics
+    // timestamp inherent, successful transfer, failed transfer
+    let best_hash = alice.client.info().best_hash;
+    let domain_block_extrinsics = alice.client.block_body(best_hash).unwrap();
+    assert_eq!(domain_block_extrinsics.unwrap().len(), 3);
+
+    // next bundle should have er which should a total of 5 trace roots
+    // timestamp + success ext + failed ext + finalize + final_state_root
+    let (_slot, bundle) = ferdie.produce_slot_and_wait_for_bundle_submission().await;
+    assert!(bundle.is_some());
+    let bundle = bundle.unwrap();
+    let er = bundle.sealed_header.header.receipt;
+
+    assert_eq!(er.execution_trace.len(), 5);
+    assert_eq!(er.execution_trace[4], er.final_state_root);
```

Comment:
Plz also adds a check to ensure the failed extrinsic is committed to the domain block extrinsic root at the expected `extrinsic_index`, this is what required by the invalid state transition fraud proof.

Reply:
isn't that implicitly checked here - https://github.com/subspace/subspace/pull/2343/files#diff-1e383587b19880e5eb1c094ad80ce370ee51977029ef8e1ae09045256df53aebR1963?

Reply:
The extrinsic is manually put into the block body by the block builder, so the check will succeed even without the fix, also we need to check the extrinsic is committed to a specific `extrinsic_index`

Reply:
PLease rephrase what you mean by commit to specific index. If you would like to me add specific check you prefer then more details on that is appraciated

Reply:
Do you mean checking if the extrinsic_data has the extrinsic at specific index ? 🤔
If so, sure I can add that

Reply:
Yeah, what I mean is the check that we have done in the verification of the invalid state transition fraud proof:
https://github.com/subspace/subspace/blob/0a51f37b2dc4b41236f215b82d48979ada0b7028/crates/sp-domains-fraud-proof/src/fraud_proof.rs#L144-L161

by using the domain block body to create a `proof_of_inclusion` of that failed extrinsic and verify the proof with the domain block extrinsic root.

Reply:
Not necessarily. Just checking the extrinsic root from the header and deriving it manually is enough imo

Reply:
Yeah, that also works 👍